content/troubleshooting/accessing-builder-machine-via-ssh.md

```diff
@@ -1,49 +1,64 @@
+
+All virtual machines are located within the private network. In order to allow users to connect to the virtual machine, Codemagic creates a temporary SSH tunnel and VNC access through the public gateway. A unique SSH key and user credentails are generated before each build which is valid for the duration of the build. The unique key and user credentials are revoked after the build is finished and cannot be reused.
 
 ## Setting up SSH connection to the virtual machine
 
-Establishing an SSH connection to the virtual machine is possible only for manually triggered builds. Setting it up requires both enabling the SSH mode in UI and running an automatically generated script in your terminal.
+Establishing an SSH connection to the virtual machine is possible only for manually triggered builds. Setting it up requires both enabling remote access in UI and running an automatically generated script in your terminal.
 
-1. In your app settings, click **Start new build**.
-2. In the **Specify build configuration** popup, check **Enable SSH access** and start the build.
-   {{< figure size="medium" src="../uploads/2019/07/enable_ssh_access.PNG" caption="" >}}
+1. Click **Start new build**.
+2. In the **Specify build configuration** popup, check **Enable remote access** and start the build.
 3. When the build has started, you will shortly see additional instructions for establishing SSH access. Run the automatically generated script in the terminal before the build finishes to establish an SSH connection to the machine running the build.
-   {{< figure size="medium" src="../uploads/2019/07/ssh_access_script.PNG" caption="" >}}
 
 If you don't run the script before the build finishes, the unique SSH key expires and can't be used anymore. A new script will be generated every time you run the build, so previous scripts cannot be reused.
 
 {{<notebox>}}
 Note that the script for establishing SSH connection works natively on Linux and MacOS, but requires additional software like Git-bash on Windows.
 {{</notebox>}}
 
-## SSH session time limit
+## Setting up VNC connection to the virtual machine
+
+Establishing a VNC connection to the virtual machine is possible only for manually triggered builds. Setting it up requires enabling remote access in UI and configuring access on your VNC client.
 
-The SSH session remains active for a maximum of 20 minutes after all build steps are completed or until the maximum build duration limit is reached, whichever comes first. If you log out from the virtual machine during an active SSH session, the build is finished automatically and you cannot access the builder via SSH for this session. You would have to run a new build to establish a new SSH connection.
+1. Click **Start new build**.
+2. In the **Specify build configuration** popup, check **Enable remote access** and start the build.
+3. Use the given **Host**, **Port** and **Username** and **Password** on your VNC client to establish the connection.
```

Comment:
I think it should be ...**Host**, **Port**, **Username** and ... instead of ...**Host**, **Port** and **Username** and....